.github/workflows/pr-container-composite.yaml

```diff
@@ -0,0 +1,229 @@
+name: 'Build and Test and PR Container'
+inputs:
+  job-name:
+    description: "Job name to display on Slack messages."
+    required: true
+    type: string
+
+  build-file:
+      description: "Path to the container file."
+      required: true
+      type: string
+
+  build-context:
+    description: "Path to the build root dir."
+    default: '.'
+    type: string
+
+  pr-filters:
+    description: "YAML dictionary of lists of path filters."
+    required: true
+    type: string
+
+  test-flags:
+    description: "Arguments passed to docker run."
+    default: ''
+    type: string
+
+  test-args:
+    description: "Arguments passed to the container."
+    default: ''
+    type: string
+
+  registry:
+    description: "URL of the container registry."
+    required: true
+    type: string
+
+  registry-user:
+    description: "Username for the container registry."
+    required: true
+    type: string
+
+  registry-repo:
+    description: "Repo within container registry project."
+    required: true
+    type: string
+
+  registry-tag-prefix:
+    description: "Prefix to use when tagging container"
+    default: ''
+    type: string
+
+  registry-tag-suffix:
+    description: "Suffix to use when tagging container"
+    default: ''
+    type: string
+
+  release-rules:
+    description: "JSON list of rule dicts for what commit types trigger releases."
+    default: |
+      [
+        {"type": "major", "release": "major"},
+        {"type": "minor", "release": "minor"},
+        {"type": "patch", "release": "patch"},
+        {"type": "no-release", "release": false},
+      
+        {"type": "chore", "release": "patch"},
+        {"type": "refactor", "release": "patch"},
+        {"type": "style", "release": "patch"},
+      
+        {"type": "docs", "release": false},
+        {"type": "test", "release": false},
+        {"type": "ci", "release": false},
+      
+        {"type": "feat", "release": "minor"},
+      
+        {"type": "revert", "release": "patch"},
+        {"type": "perf", "release": "patch"},
+        {"type": "fix", "release": "patch"},
+        {"type": "build", "release": "patch"},
+      ]
+    type: string
+
+  release-branches:
+    description: "JSON list of regex branch filters."
+    default: |
+      [
+        '+([0-9])?(.{+([0-9]),x}).x',
+        'main'
+      ]
+    type: string
+
+  release-tag-format:
+    description: "Semantic-release Github release tag format."
+    default: '${version}'
+    type: string
+
+  slack-channel:
+    description: "ID of the Slack channel to post updates to."
+    required: true
+    type: string
+
+  status-failure:
```

Comment:
We can drop `status-failure` as an argument. We'll rework slack errors towards the end of getting this working. Keep the slack channel and token args for now as we will eventually use them.